plugins/inventory_sync/controller.ts

```diff
@@ -90,7 +89,7 @@ export class ControllerPlugin extends lib.BaseControllerPlugin {
 		if (
 			!acquisitionRecord
 			|| acquisitionRecord.instanceId === instanceId
-			|| !this.controller.instances.has(acquisitionRecord.instanceId)
+			|| !this.controller.instances!.has(acquisitionRecord.instanceId)
```

Comment:
It shouldn't be an optional type, but it used to wipe the data stored when an error happened during startup. Should we be saving anything if startup fails? Probably not. It would be best to load this data before creating the Controller instance so there's no chance of it saving an empty map. But that's something for another pull request.